clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractPartitionAssignor.java

```diff
@@ -87,4 +88,10 @@ public void onAssignment(Assignment assignment) {
         list.add(value);
     }
 
+    protected static List<TopicPartition> partitions(String topic, int numPartitions) {
+        List<TopicPartition> partitions = new ArrayList<>();
```

Comment:
You can pass `numPartitions` to the `ArrayList` constructor so that the internal array is allocated with the right size.